ctapipe/io/simteleventsource.py

```diff
@@ -1012,38 +1016,59 @@ def _parse_simulation_header(self):
         )
 
         for tel_id in self.file_.pixel_monitorings.keys():
-            nsb_rate = (self.file_.pixel_monitorings.get(tel_id, {})
-                   .get("nsb_rate", None)
-                    )
-            qe_rel = (
-                    self.file_.pixel_monitorings
-                    .get(tel_id, {})
-                    .get("qe_rel", None)
-                        )
-            hv_rel = (
-                    self.file_.pixel_monitorings
-                    .get(tel_id, {})
-                    .get("hv_rel", None)
-                        )
-            current = (
-                    self.file_.pixel_monitorings
-                    .get(tel_id, {})
-                    .get("current", None)
-                        )
-            fadc_amp_hg = (
-                    self.file_.pixel_monitorings
-                    .get(tel_id, {})
-                    .get("fadc_amp_hg", None)
-                        )
-
-            service_container = SimulatedPixelMonitoring(
+            nsb_rate = (self.file_.pixel_monitorings.get(tel_id, {}).get("nsb_rate", None))
+            qe_rel = (self.file_.pixel_monitorings.get(tel_id, {}).get("qe_rel", None))
+            hv_rel = (self.file_.pixel_monitorings.get(tel_id, {}).get("hv_rel", None))
+            current = (self.file_.pixel_monitorings.get(tel_id, {}).get("current", None))
+            fadc_amp_hg = (self.file_.pixel_monitorings.get(tel_id, {}).get("fadc_amp_hg", None))
+            
+            service_container = PixelMonitoringContainer(
                     nsb_pe_rate=nsb_rate,
                     qe_rel=qe_rel,
                     high_voltage=hv_rel,
                     current=current,
                     fadc_amp=fadc_amp_hg,
                     )
-            simulation_config.tel[tel_id] = service_container 
+
+            calib = (self.file_.laser_calibrations.get(tel_id, {}).get("calib", None))
+            max_int_frac = (self.file_.laser_calibrations.get(tel_id, {}).get("max_int_frac", None))
+            max_pix_tm_frac = (self.file_.laser_calibrations.get(tel_id, {}).get("max_pix_tm_frac", None))
+            tm_calib = (self.file_.laser_calibrations.get(tel_id, {}).get("tm_calib", None))
+            flat_fielding = (self.file_.laser_calibrations.get(tel_id, {}).get("flat_fielding", None))
+
+            laser_container = LaserCalibrationContainer(
+                    calib=calib,
+                    max_int_frac=max_int_frac,
+                    max_pixtm_frac=max_pix_tm_frac,
+                    tm_calib=tm_calib,
+                    flat_fielding=flat_fielding,
+                    )
+
+            monitor_id = (self.file_.camera_monitorings.get(tel_id, {}).get("monitor_id", None))
```

Comment:
You can just directly put it into the container:

LaserCalibrationContainer(
    calib=self.file.get....
)